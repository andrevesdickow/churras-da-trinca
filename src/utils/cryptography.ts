import crypto from 'crypto';
import { isEmpty } from 'lodash';

function getBytes(str: string) {
  const bytes = [];
  for (let i = 0; i < str.length; ++i) {
    bytes.push(str.charCodeAt(i));
  }
  return bytes;
}

/**
 * Criptografa um texto
 * @param textToEncrypt texto
 * @param secretKey chave secreta
 * @returns texto criptografado
 */
export function encrypt(textToEncrypt: string, secretKey: string) {
  if (isEmpty(textToEncrypt) || isEmpty(secretKey)) {
    return '';
  }

  const key = Buffer.from(getBytes(secretKey));
  const iv = Buffer.from([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]);

  const cipher = crypto.createCipheriv('aes-128-cbc', key, iv);
  let encrypted = cipher.update(textToEncrypt);
  encrypted = Buffer.concat([encrypted, cipher.final()]);
  return encrypted.toString('base64');
}
