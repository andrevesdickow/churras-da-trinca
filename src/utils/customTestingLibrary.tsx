import '../app/global.css';
import React from 'react';
import { ToastContainer } from '@/components/Toast';
import { NextThemeProvider } from '@/providers/NextThemeProvider';
import { Provider as TooltipProvider } from '@radix-ui/react-tooltip';
import { render } from '@testing-library/react';

const customRender = (ui: React.ReactElement, { ...renderOptions } = {}) => render(
  <NextThemeProvider>
    <TooltipProvider>
      {ui}
      <ToastContainer />
    </TooltipProvider>
  </NextThemeProvider>,
  renderOptions
);

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { customRender as render };
