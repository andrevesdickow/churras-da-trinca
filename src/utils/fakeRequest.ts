export function fakeRequest<T = any>(delay: number, value: T) {
  return new Promise((resolve) => setTimeout(resolve, delay, value));
}
