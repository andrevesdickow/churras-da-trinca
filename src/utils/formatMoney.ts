/**
 * Formata valor monetÃ¡rio para real brasileiro
 * @param value valor
 * @returns Valor formatado -> R$ 1.000,00
 */
export function formatMoney(value: number) {
  return new Intl.NumberFormat('pt-BR', {
    style: 'currency',
    currency: 'BRL'
  }).format(value);
}
