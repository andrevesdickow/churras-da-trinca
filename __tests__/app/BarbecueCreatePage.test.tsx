import BarbecueCreatePage from '@/app/churras/criar/page';
import { fireEvent, render, screen, waitForElementToBeRemoved } from '@/utils/customTestingLibrary';

const replace = jest.fn();

jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      replace
    };
  }
}));

describe('<BarbecueCreatePage />', () => {
  it('should render correctly Barbecue Create Page', () => {
    const { container } = render(<BarbecueCreatePage />);

    expect(screen.getByText(/Churras da TRINCA/i)).toBeInTheDocument();

    expect(screen.getByText(/Criar churras/i)).toBeInTheDocument();
    expect(screen.getByRole('link', { name: 'Voltar' })).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should click button correctly submit form in Barbecue Create Page', async () => {
    render(<BarbecueCreatePage />);

    fireEvent.submit(screen.getByRole('button'));

    expect(await screen.findByRole('alert')).toBeInTheDocument();
    expect(await screen.findByText('Descrição é obrigatório')).toBeInTheDocument();

    fireEvent.input(screen.getByRole('textbox', { name: 'Descrição' }), { target: { value: 'Teste' } });

    await waitForElementToBeRemoved(() => screen.queryByRole('alert'));
  });

  it('should render back button correctly in Barbecue Create Page', () => {
    render(<BarbecueCreatePage />);

    expect(screen.getByRole('link', { name: 'Voltar' })).toHaveAttribute('href', '/');
  });
});
