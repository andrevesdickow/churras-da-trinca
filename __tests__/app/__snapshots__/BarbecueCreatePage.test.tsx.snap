// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BarbecueCreatePage /> should render correctly Barbecue Create Page 1`] = `
<div>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <form
    class="flex flex-col gap-3 p-8 rounded-3xl z-[1px] min-w-[400px] w-full h-max bg-slate-50 dark:bg-slate-800 shadow-xl"
  >
    <h3
      class="text-center font-bold text-amber-900 dark:text-slate-50"
    >
      Churras da TRINCA
    </h3>
    <div
      class="grid md:grid-cols-2 grid-cols-1 gap-4"
    >
      <div
        class="relative z-0 mb-4"
      >
        <input
          class="block py-2.5 px-0 w-full text-sm text-slate-900 bg-transparent border-0 border-b-2 border-slate-500 appearance-none dark:border-slate-300 dark:focus:border-amber-400 dark:text-slate-100 focus:outline-none focus:ring-0 focus:border-amber-400 peer"
          id="date"
          min="2023-10-16"
          name="date"
          placeholder=" "
          role="textbox"
          type="date"
          value="2023-10-16"
        />
        <label
          class="absolute text-sm text-slate-500 dark:text-slate-300 duration-300 transform -translate-y-6 scale-75 top-3 -z-10 origin-[0] peer-focus:left-0 peer-focus:text-amber-400 peer-focus:dark:text-amber-400 peer-placeholder-shown:scale-100 peer-placeholder-shown:translate-y-0 peer-focus:scale-75 peer-focus:-translate-y-6"
          for="date"
        >
          Data
        </label>
      </div>
      <div
        class="relative z-0 mb-4"
      >
        <input
          class="block py-2.5 px-0 w-full text-sm text-slate-900 bg-transparent border-0 border-b-2 border-slate-500 appearance-none dark:border-slate-300 dark:focus:border-amber-400 dark:text-slate-100 focus:outline-none focus:ring-0 focus:border-amber-400 peer"
          id="description"
          name="description"
          placeholder=" "
          role="textbox"
          type="text"
          value=""
        />
        <label
          class="absolute text-sm text-slate-500 dark:text-slate-300 duration-300 transform -translate-y-6 scale-75 top-3 -z-10 origin-[0] peer-focus:left-0 peer-focus:text-amber-400 peer-focus:dark:text-amber-400 peer-placeholder-shown:scale-100 peer-placeholder-shown:translate-y-0 peer-focus:scale-75 peer-focus:-translate-y-6"
          for="description"
        >
          Descrição
        </label>
      </div>
      <div
        class="relative z-0 mb-4"
      >
        <input
          class="block py-2.5 px-0 w-full text-sm text-slate-900 bg-transparent border-0 border-b-2 border-slate-500 appearance-none dark:border-slate-300 dark:focus:border-amber-400 dark:text-slate-100 focus:outline-none focus:ring-0 focus:border-amber-400 peer"
          id="priceWithoutDrink"
          name="priceWithoutDrink"
          placeholder=" "
          type="tel"
          value="0,00"
        />
        <label
          class="absolute text-sm text-slate-500 dark:text-slate-300 duration-300 transform -translate-y-6 scale-75 top-3 -z-10 origin-[0] peer-focus:left-0 peer-focus:text-amber-400 peer-focus:dark:text-amber-400 peer-placeholder-shown:scale-100 peer-placeholder-shown:translate-y-0 peer-focus:scale-75 peer-focus:-translate-y-6"
          for="priceWithoutDrink"
        >
          Valor de contribuição (sem bebida)
        </label>
      </div>
      <div
        class="relative z-0 mb-4"
      >
        <input
          class="block py-2.5 px-0 w-full text-sm text-slate-900 bg-transparent border-0 border-b-2 border-slate-500 appearance-none dark:border-slate-300 dark:focus:border-amber-400 dark:text-slate-100 focus:outline-none focus:ring-0 focus:border-amber-400 peer"
          id="priceWithDrink"
          name="priceWithDrink"
          placeholder=" "
          type="tel"
          value="0,00"
        />
        <label
          class="absolute text-sm text-slate-500 dark:text-slate-300 duration-300 transform -translate-y-6 scale-75 top-3 -z-10 origin-[0] peer-focus:left-0 peer-focus:text-amber-400 peer-focus:dark:text-amber-400 peer-placeholder-shown:scale-100 peer-placeholder-shown:translate-y-0 peer-focus:scale-75 peer-focus:-translate-y-6"
          for="priceWithDrink"
        >
          Valor de contribuição (com bebida)
        </label>
      </div>
    </div>
    <div
      class="relative z-0 mb-4"
    >
      <textarea
        class="block py-2.5 px-0 w-full text-sm text-slate-900 bg-transparent border-0 border-b-2 border-slate-500 appearance-none dark:border-slate-300 dark:focus:border-amber-400 dark:text-slate-100 focus:outline-none focus:ring-0 focus:border-amber-400 peer"
        id="additionalObservations"
        name="additionalObservations"
        placeholder=" "
        rows="4"
      />
      <label
        class="absolute text-sm text-slate-500 dark:text-slate-300 duration-300 transform -translate-y-6 scale-75 top-3 -z-10 origin-[0] peer-focus:left-0 peer-focus:text-amber-400 peer-focus:dark:text-amber-400 peer-placeholder-shown:scale-100 peer-placeholder-shown:translate-y-0 peer-focus:scale-75 peer-focus:-translate-y-6"
        for="additionalObservations"
      >
        Observações adicionais
      </label>
    </div>
    <button
      class="inline-flex items-center justify-center gap-2 rounded-3xl text-sm font-medium uppercase ring-offset-white transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-violet-950 focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 dark:ring-offset-violet-950 dark:focus-visible:ring-violet-300 bg-slate-950 text-white hover:bg-slate-950/90 dark:bg-slate-950 dark:text-white dark:hover:bg-slate-950/50 h-10 px-4 py-2"
      type="submit"
    >
      Criar churras
    </button>
    <a
      class="text-amber-400 hover:text-amber-500 transition-colors text-center text-sm"
      href="/churras"
    >
      Voltar
    </a>
  </form>
</div>
`;
